plugin.video.montreal.greek-tv/resources/lib/default.py

```diff
@@ -0,0 +1,96 @@
+# -*- coding: utf-8 -*-
+
+import urllib.parse, os, sys
+import xbmcvfs, xbmcgui, xbmcplugin, xbmcaddon
+
+addon_id       = 'plugin.video.montreal.greek-tv'
+Home           = xbmcvfs.translatePath(xbmcaddon.Addon().getAddonInfo('path'))
+art            = xbmcvfs.translatePath(Home + '/resources/art/')
+icon           = xbmcvfs.translatePath(os.path.join(Home, 'icon.png'))
+tv_icon        = xbmcvfs.translatePath(os.path.join(art , 'montreal_tv.png'))
+radio_icon     = xbmcvfs.translatePath(os.path.join(art , 'montreal_radio.png'))
+radio_fanart   = xbmcvfs.translatePath(os.path.join(art , 'montreal_backround.jpg'))
+
+__handle__     = int(sys.argv[1])
+
+TV_URL = 'http://live.greektv.ca/hls1/greektv.m3u8'
+RADIO_URL = 'http://live.greekradio.ca:8000/live'
+
+def Index():
+    addDir("Montreal Greek TV", TV_URL, 2, tv_icon, radio_fanart)
+    addDir("Montreal Greek Radio", RADIO_URL, 2, radio_icon, radio_fanart)
+    xbmcplugin.endOfDirectory(__handle__)
+
+def addDir(name, url, mode, iconimage, fanart):
+        u = sys.argv[0]+"?url="+urllib.parse.quote_plus(url)+"&mode="+str(mode)+"&name="+urllib.parse.quote_plus(name)+"&iconimage="+urllib.parse.quote_plus(iconimage)+"&fanart="+urllib.parse.quote_plus(fanart)
+        liz = xbmcgui.ListItem(name)
+        liz.setArt({'thumb': iconimage, "fanart": fanart})
+        liz.setInfo(type = "Video", infoLabels = {"Title": name, "Plot": "" } )
+        liz.setProperty('IsPlayable', 'true')
+        xbmcplugin.addDirectoryItem(handle = __handle__, url = u, listitem = liz,isFolder = False)
+
+def Playlinks(name, url):
+        liz = xbmcgui.ListItem(name)
+        liz.setProperty('isFolder', 'false')
+        liz.setArt({'thumb': radio_icon})
+        liz.setInfo('video', {'Title': '', 'Plot': ''})
+        liz.setProperty('IsPlayable','true')
+        liz.setPath(url)
+        xbmcplugin.setResolvedUrl(__handle__, True, liz)
+        
+def get_params():
```

Comment:
You might want to look into the routing script module as it is much more clean than this (not mandatory)

Reply:
new update i have adapted the routing script